webapp-django/crashstats/signature/views.py

```diff
@@ -35,36 +37,36 @@
 )
 
 
-def get_validated_params(request):
-    params = get_params(request)
-    if isinstance(params, http.HttpResponseBadRequest):
-        # There was an error in the form, let's return it.
-        return params
+def get_validated_params(view):
+    @functools.wraps(view)
+    def inner(request, *args, **kwargs):
+        try:
+            params = get_params(request)
 
-    if len(params['signature']) > 1:
-        return http.HttpResponseBadRequest(
-            'Invalid value for "signature" parameter, '
-            'only one value is accepted'
-        )
+            if len(params['signature']) > 1:
+                raise ValidationError(
+                    'Invalid value for "signature" parameter, '
+                    'only one value is accepted'
+                )
 
-    if not params['signature'] or not params['signature'][0]:
-        return http.HttpResponseBadRequest(
-            '"signature" parameter is mandatory'
-        )
+            if not params['signature'] or not params['signature'][0]:
+                raise ValidationError(
+                    '"signature" parameter is mandatory'
+                )
+        except ValidationError, e:
+            return http.HttpResponseBadRequest(str(e))
 
-    return params
+        kwargs['params'] = params
+        return view(request, *args, **kwargs)
```

Comment:
A note about the use of `+= (params,)` is that suppose you have a view that is defined to take in an ID or something in the `urls.py` then it would end up looking like this:

```python
@get_validated_params
def my_view(request, id, params):
```